net/ip.c

```diff
@@ -0,0 +1,448 @@
+		 (int)na->nip_port);
+
+	if (na->nip_format == M0_NET_IP_LNET_FORMAT)
+		M0_LOG(M0_DEBUG, "type=%d portal=%d tmid=%d autotm=%s",
+		       (int)na->fmt_pvt.la.nla_type,
+		       (int)na->fmt_pvt.la.nla_portal,
+		       (int)na->fmt_pvt.la.nla_tmid,
+		       na->fmt_pvt.la.nla_autotm ? "true" : "false");
+	else
+		M0_LOG(M0_DEBUG, "family=%d type=%d",
+		       (int)na->fmt_pvt.ia.nia_family,
+		       (int)na->fmt_pvt.ia.nia_type);
+
+	if (na->nip_format == M0_NET_IP_LNET_FORMAT) {
+		rc = na->fmt_pvt.la.nla_autotm ?
+		     snprintf(tmid, ARRAY_SIZE(tmid), "*") :
+		     snprintf(tmid, ARRAY_SIZE(tmid), "%d",
+			      na->fmt_pvt.la.nla_tmid);
+		M0_ASSERT(rc < ARRAY_SIZE(tmid));
+		inet_ntop(AF_INET, &na->ip_n.sn[0], ip_p, ARRAY_SIZE(ip_p));
+		rc = snprintf(buf, M0_NET_IP_STRLEN_MAX,
+			      "%s@%s:12345:%d:%s",
+			      na->fmt_pvt.la.nla_type == M0_NET_IP_PROTO_LO ?
+			      "0" : ip_p,
+			      na->fmt_pvt.la.nla_type ==
+			      M0_NET_IP_PROTO_LO ? "lo" :
+			      ((na->fmt_pvt.la.nla_type ==
+			      M0_NET_IP_PROTO_TCP) ? "tcp": "o2ib"),
+			      na->fmt_pvt.la.nla_portal, tmid);
+	} else if (na->nip_format == M0_NET_IP_INET_IP_FORMAT) {
+		if (na->fmt_pvt.ia.nia_family != M0_NET_IP_AF_UNIX) {
+			inet_ntop(na->fmt_pvt.ia.nia_family ==
+				  M0_NET_IP_AF_INET ? AF_INET : AF_INET6,
+				  &na->ip_n.sn[0], ip_p, ARRAY_SIZE(ip_p));
+			rc = snprintf(buf, M0_NET_IP_STRLEN_MAX, "%s:%s:%s@%d",
+				      na->fmt_pvt.ia.nia_family ==
+				      M0_NET_IP_AF_INET ? "inet" : "inet6",
+				      ip_protocol[na->fmt_pvt.ia.nia_type],
+				      ip_p, na->nip_port);
+		} else
+			M0_LOG(M0_ERROR, "Format is currently not supported");
+	} else if (na->nip_format == M0_NET_IP_INET_HOSTNAME_FORMAT) {
+		inet_ntop(AF_INET, &na->ip_n.sn[0], ip_p, ARRAY_SIZE(ip_p));
+		m0_net_ip_to_hostname(ip_p, hostname);
+		rc = snprintf(buf, M0_NET_IP_STRLEN_MAX, "inet:%s:%s@%d", 
+			      ip_protocol[na->fmt_pvt.ia.nia_type], hostname,
+			      na->nip_port);
+	}
+	if (rc >= M0_NET_IP_STRLEN_MAX)
+		M0_LOG(M0_ERROR, "Name too long. Reqd: %d buf_len: %d", rc,
+		       M0_NET_IP_STRLEN_MAX);
+	M0_LOG(M0_DEBUG, "Address constructed: %s", buf);
+
+	return 0;
+}
+
+M0_INTERNAL int m0_net_hostname_to_ip(char *hostname, char *ip,
+				      enum m0_net_ip_format *fmt)
+{
+	struct hostent  *hname;
+	struct in_addr **addr;
+	uint32_t         ip_n[4];
+	int              i;
+	int              n;
+	char            *cp;
+	char             name[M0_NET_IP_STRLEN_MAX] = {};
+
+	M0_ENTRY("Hostname=%s", (char*)hostname);
+	cp = strchr(hostname, '@');
+	if (cp == NULL)
+		return M0_ERR(-EINVAL);
+
+	n = cp - hostname;
+	name[n] = '\0';
+	strncat(name, hostname, n);
+
+	if (inet_pton(AF_INET, name, &ip_n[0]) == 1 ||
+	    inet_pton(AF_INET6, name, &ip_n[0]) == 1) {
+		/* Copy ip address as it is. */
+		*fmt = M0_NET_IP_INET_IP_FORMAT;
+		strcpy(ip, name);
+	} else {
+		*fmt = M0_NET_IP_INET_HOSTNAME_FORMAT;
+		if ((hname = gethostbyname(name)) == NULL) {
```

Comment:
As per the coding style, "control flow statement conditions ought to have no side-effects".

Reply:
Fixed.